src/sagas.js

```diff
@@ -802,6 +808,60 @@ function* removeVm (action) {
   yield stopProgress({ vmId: action.payload.vmId, name: 'remove', result })
 }
 
+function* fetchAllEvents (action) {
+  const { userId } = action.payload
+  const events = yield callExternalAction('events', Api.events, { payload: {} })
+
+  if (events.error) {
+    return
+  }
+
+  const internalEvents = events.event
+    ? events.event
+      .filter((event) =>
+        event.severity === 'error' &&
+        event.user &&
+        event.user.id === userId
+      )
+      .map((event) => Api.eventToInternal({ event }))
+    : []
+  yield put(setUserMessages({ messages: internalEvents }))
+}
+
+function* dismissEvent (action) {
+  const { event } = action.payload
+  if (event.get('id')) {
+    const result = yield callExternalAction('dismissEvent', Api.dismissEvent, action)
+
+    if (result.status === 'complete') {
+      yield fetchAllEvents(action)
+    }
+  } else {
+    yield put(dismissUserMessage({ time: event.get('time') }))
+  }
+}
+
+function* clearEvents (action) {
+  const events = yield callExternalAction('events', Api.events, { payload: {} })
```

Comment:
Because now for notifications engine is source of truth and that's why all information should be checked on engine. VM Portal has just copy of that information nothing more.

Reply:
So the intent is to clear all events, even the ones that have not been shown to the user yet?

Reply:
Yes, button `Clear All`, clear all. =) Because events generated by VM Portal already shown to user, so I don't see a problem to clear all error events.